### PR TITLE
Improvements

### PR DESCRIPTION
**Pull request by benjaminaudet**
Sorry for the name of the branch, it's a mistake.

So, on this PR 🚀 :

* Some cleaning
* Add CSV export
* Add host to .env
* Update to bimdata_api_client v3 
* No more generated files on origin
* An installer (install.py) that install required modules + the connector
* Setting up the host in Power BI parameters
* Manage multi-models

### CODE REVIEW

File: dashboard/src/get_elements.py [hunk 1]
```diff
@@ -34,44 +37,61 @@
             You can specify property names to ONLY includes in the includes list
             This is working with XOR logic, if there are elements in includes, excludes will not be considered
 '''
+
+def smart_cast(value):
```
Comment (Amoki) [inline dashboard/src/get_elements.py:41]: still used ?

File: dashboard/src/__init__.py [hunk 2]
```diff
@@ -1,66 +1,187 @@
...
+
+def get_ifc_window(debug_type, excludes=[], includes=[]):
+    ifc_type = 'IfcWindow'
+    get_elements_window = GetElements(ifc_type=ifc_type, debug=debug_type, properties_options={'excludes': excludes, 'includes': includes})
+    print_begin(ifc_type)
+    Elements = get_elements_window.run()
+    print_end(ifc_type)
+    return Elements
+
+def get_ifc_zone(debug_type, excludes=[], includes=[]):
```
Comment (Amoki) [inline dashboard/src/__init__.py]: having the same method copy-pasted 10 times is never a good thing. You can factorize them

File: dashboard/src/get_elements.py [hunk 3]
```diff
@@ -5,11 +5,14 @@
...
 import logging
-
-
+from decimal import *
+from collections import Counter
+import time
 import pprint
 pp = pprint.PrettyPrinter(indent=2, width=120)
 
+getcontext().prec = 3
```
Comment (Amoki) [inline dashboard/src/get_elements.py]: 3 = micro-meter or millimeter?
We would need micro-meter

File: dashboard/src/get_elements.py [hunk 4]
```diff
@@ -34,44 +37,61 @@
...
+            print(self.ifc_pks)
+            self.host = os.getenv('HOST')
 
     def config(self):
         configuration = bimdata_api_client.Configuration()
-        configuration.api_key['Authorization'] = self.access_token
-        configuration.api_key_prefix['Authorization'] = 'Bearer'
-        configuration.host = 'https://api-staging.bimdata.io'
+        configuration.access_token = self.access_token
+        configuration.host = f'https://api-{self.host}.bimdata.io'
```
Comment (Amoki) [inline dashboard/src/get_elements.py]: host should be the API url directly. We'll remove the '-beta' in the prod url soon

File: dashboard/src/get_elements.py [hunk 5]
```diff
@@ -98,19 +118,25 @@ def get_properties_from_elements(self):
                 for prop in pset['properties']:
                     for prop_target in element['properties']:
                         if prop_target['name'] == prop['definition']['name']:
-                            prop_target['value'] = prop['value'] if prop['value'] not in ['', None] else ''
+                            value = prop['value'] # if prop['value'] not in ['', None] else 'yolo'
```
Comment (Amoki) [inline dashboard/src/get_elements.py]: ?

File: dashboard/src/get_elements.py [hunk 6]
```diff
@@ -138,29 +164,50 @@ def raw_elements_to_elements(self, api_response):
...
             self.elements = [elem for elem in self.elements.values() if elem['type'] == self.ifc_type]
             return
         self.elements = [elem for elem in self.elements.values()]
 
+    def define_column_types(self):
+        for key in self.types_ref:
+            c = Counter(self.types_ref[key])
+            self.df = self.df.astype({key: c.most_common()[0][0]})
+
+    def fill_void(self):
```
Comment (Amoki) [inline dashboard/src/get_elements.py:178]: Why do we need to fill void? 0.0 is very different than no value

File: dashboard/src/get_elements.py [hunk 7]
```diff
@@ -138,29 +164,50 @@ def raw_elements_to_elements(self, api_response):
...
         try:
-            api_response = ifc_api.get_raw_elements(self.cloud_pk, self.ifc_pk, self.project_pk)
-            self.raw_elements_to_elements(api_response)
-            self.filter_by_types()
+            for ifc_pk in self.ifc_pks:
+                print(f'retrieving... {ifc_pk}')
+                api_response = ifc_api.get_raw_elements(self.cloud_pk, ifc_pk, self.project_pk, type=self.ifc_type)
+                self.raw_elements_to_elements(api_response)
+                print(f'good for {ifc_pk}')
+                # self.filter_by_types()
```
Comment (Amoki) [inline dashboard/src/get_elements.py]: dead code

**Review by Amoki — CHANGES_REQUESTED** [review]

File: dashboard/src/get_elements.py (hunk 1 above)
Comment (benjaminaudet) [reply dashboard/src/get_elements.py:41]: Not dead code, still WIP

File: dashboard/src/get_elements.py (hunk 6 above)
Comment (benjaminaudet) [reply dashboard/src/get_elements.py:178]: It was in the type casting process, when it was float it needed to have 0.0 assigned

File: dashboard/src/get_elements.py (hunk 7 above)
Comment (benjaminaudet) [reply dashboard/src/get_elements.py]: Not dead code, still WIP

File: dashboard/src/get_elements.py (hunk 5 above)
Comment (benjaminaudet) [reply dashboard/src/get_elements.py]: I'm asking the same question to myself

File: dashboard/src/get_elements.py (hunk 4 above)
Comment (benjaminaudet) [reply dashboard/src/get_elements.py]: done

**Pull request merged**